core/database.ml

```diff
@@ -2,11 +2,13 @@ open CommonTypes
 open Utility
 
 let connection_info
-  = Settings.(option "database_args"
-              |> synopsis "Database host, name, user, and password"
+  = let setting =
+      Settings.(option "database_args"
+              |> synopsis "Database host, port, user, and password"
               |> to_string from_string_option
-              |> convert Utility.some
+              |> convert (fun s -> Utility.some (Sys.expand s))
```

Comment:
This is a bit ad hoc.  Are there any other places where environment variable expansion would be helpful?  Maybe the fact that environment vars are expanded for this specific option (and the reason why) should be documented in the synopsis or a comment?

Reply:
We do the same thing in places where filenames/pathnames come up, e.g. `parse.ml`, `moduleUtils.ml`, etc.

Reply:
Actually, I see that past-me has defined a utility function for this purpose called `parse_paths` in `settings.ml:128`. However, it converts colon/comma separated string into a list of strings (each expanded). I could use this. It will require me to patch up the places where we parse the database arguments string, though, it might be better to parse it up front with `parse_paths` (maybe rename this function too).

Reply:
I think my suggestion has to be implemented as a separate patch. See #970.